virtual_rainforest/core/config.py

```diff
@@ -0,0 +1,254 @@
+            return None
+
+        # Track down all toml files in the config folder
+        files = []
+        for file in os.listdir(filepath):
+            if file.endswith(".toml"):
+                files.append(file)
+    else:
+        # Loop to check that all specified files can be found
+        not_found = []
+        for file in in_files:
+            if not os.path.isfile(f"{filepath}/{file}"):
+                not_found.append(file)
+
+        if len(not_found) != 0:
+            LOGGER.critical(
+                f"The files the user specified to be read from are not all found in "
+                f"{filepath}. The following files are missing:\n{not_found}"
+            )
+            return None
+
+        files = in_files
+
+    # Load all toml files that we want to read from
+    for file in files:
+        # If not then read in the file data
+        with open(os.path.join(filepath, file), "rb") as f:
+            try:
+                toml_dict = tomllib.load(f)
+                # Check for repeated entries across previous nested dictionaries
+                for item in file_data:
+                    repeats = check_dict_leaves(item[1], toml_dict, [])
+                    for elem in repeats:
+                        conflicts.append((elem, file, item[0]))
+
+                file_data.append((file, toml_dict))
+            except tomllib.TOMLDecodeError as err:
+                LOGGER.critical(
+                    f"Configuration file {file} is incorrectly formatted.\n"
+                    f"Failed with the following message:\n{err}"
+                )
+                return None
+
+    # Check if any tags are repeated across files
+    if len(conflicts) != 0:
+        # If so generate full list of errors
+        msg = "The following tags are defined in multiple config files:\n"
+        for conf in conflicts:
+            msg += f"{conf[0]} defined in both {conf[1]} and {conf[2]}\n"
+        msg = msg[:-1]
+        LOGGER.critical(msg)
+        return None
+
+    # Merge everything into a single dictionary
+    config_dict: dict = {}
+    for item in file_data:
+        dpath.util.merge(config_dict, item[1])
+
+    # Find which other schema should be searched for
+    try:
+        modules = config_dict["config"]["core"]["modules"]
+    except KeyError:
+        LOGGER.critical(
+            "Core configuration does not specify which other modules should be "
+            "configured!"
+        )
+        return None
+
+    # Add core to list of modules if its not already included
+    if "core" not in modules:
+        modules.append("core")
+
+    if len(modules) != len(set(modules)):
+        LOGGER.critical(
+            f"The list of modules to configure given in the core configuration file "
+            f"repeats {len(modules) - len(set(modules))} names!"
+        )
+        return None
+
+    # Construct combined schema for all relevant modules
+    comb_schema: dict = {}
```

Comment:
For this you might want to consider [ChainMap](https://docs.python.org/3/library/collections.html#collections.ChainMap), as a more elegant solution to combine python dictionaries.